.github/workflows/nodejs.yml

```diff
@@ -0,0 +1,17 @@
+name: Node.js CI
+on: [push, pull_request]
```

Comment:
Two things here:

1. I think only using `pull_request` is better. It will still trigger when you push on a branch that is part of a PR and that way you don't run tests on every commit push before that (which is annoying for some people to get failed run notifications when they are doing some steps that require to break tests for certain time).
2. Based on that, I like to name the workflow name and file name based on when it triggers. For example, `Pull request` and `pull-request.yml`.

Reply:
Presumably breaking tests for a certain time should only ever occur on a feature branch, not on master? In that case I suggest:

```yaml
on:
  push:
    branches: [ master ]
  pull_request:
```

Reply:
You should never push into `master` directly I would say, thus the PR checks it and only allows merges that pass the CI. Am I missing something ?

Reply:
You could have two pull requests which *individually* pass the CI (because neither introduces a change that beaks `master`), but that interact in such a way that (even though there are no merge conflicts) `master` breaks after both have been merged.

Reply:
Right, that's why I force Github to have all branches updated before merging. But I'm fine with the config you suggested 👍

Reply:
Using branch protection to enforce branches be up-to-date and pass tests before merging is of course the best solution :+1: